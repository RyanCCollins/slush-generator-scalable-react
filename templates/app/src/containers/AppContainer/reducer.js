import * as types from './constants';

export const initialState = {
  isMobile: false,
};

const appContainerReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.DEFAULT_ACTION:
        return state;
      default:
        return state;
    }
  };

export default appContainerReducer;
