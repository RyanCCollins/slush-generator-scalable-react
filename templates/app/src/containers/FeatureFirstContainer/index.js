import React, { PropTypes, Component } from 'react';
/* eslint-disable import/no-unresolved */
import { FeatureFirstComponent } from 'components';
/* eslint-enable import/no-unresolved */
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as MyActions from './actions';

class FeatureFirstContainer extends Component {
  constructor(props) {
    super(props);
    this.initiateLoading = this.initiateLoading.bind(this);
  }
  componentDidMount() {
    this.initiateLoading();
  }
  initiateLoading() {
    const {
      actions,
    } = this.props;
    actions.loadDataInitiation();
  }
  render() {
    const {
      isLoading,
    } = this.props;
    return (
      <div>
        {isLoading ?
          <h1>LOADING...</h1>
        :
          <FeatureFirstComponent />
        }
      </div>
    );
  }
}

FeatureFirstContainer.propTypes = {
  actions: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
});

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators(MyActions, dispatch),
});

const StyledContainer = cssModules(FeatureFirstComponent, styles);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StyledContainer);
