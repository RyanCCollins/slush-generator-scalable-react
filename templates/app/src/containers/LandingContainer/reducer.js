import update from 'react-addons-update';
import * as types from './constants';

export const initialState = {
  isLoading: false,
  name: null,
  isShowingModal: false,
  contributors: [
    {
      name: 'Ryan Collins',
      github: 'ryanccollins',
      bio: 'Experienced Software Engineer specializing in implementing cutting-edge ' +
        'technologies in a multitude of domains, focusing on React and Front End.  ' +
        'Weekend Data Scientist and Functional Programmer.',
      avatar: 'https://avatars.githubusercontent.com/u/13810084?v=3',
    },
    {
      name: 'Andreas Daiminger',
      github: 'adai183',
      bio: 'I started to code about 2 years ago and have been very ' +
        'focused on developing cutting edge UI components.' +
        'I have a passion for functional programming, and I love creating ' +
        'environments that make it easier for developers to write consistent, testable code.',
      avatar: 'https://avatars.githubusercontent.com/u/13679375?v=3',
    },
  ],
};

const landingReducer =
  (state = initialState, action) => {
    switch (action.type) {
      case types.OPEN_MODAL:
        return update(state, {
          isShowingModal: {
            $set: true,
          },
        });
      case types.CLOSE_MODAL:
        return update(state, {
          isShowingModal: {
            $set: false,
          },
        });
      case types.LOAD_DATA_INITIATION:
        return update(state, {
          isShowingModal: {
            $set: false,
          },
          isLoading: {
            $set: true,
          },
          name: {
            $set: action.name,
          },
        });
      case types.LOAD_DATA_SUCCESS:
        return update(state, {
          isLoading: {
            $set: false,
          },
        });
      default:
        return state;
    }
  };

export default landingReducer;
