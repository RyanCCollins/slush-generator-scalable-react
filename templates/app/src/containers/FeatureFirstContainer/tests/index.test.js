import FeatureFirstContainer from '../index';
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState as featureComponent } from '../reducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<FeatureFirstContainer />', () => {
  it('renders with default props', () => {
    const store = mockStore({ featureComponent });
    const wrapper = shallow(
      <FeatureFirstContainer store={store} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
