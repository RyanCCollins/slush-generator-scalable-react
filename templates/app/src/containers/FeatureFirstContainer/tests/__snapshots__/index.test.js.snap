exports[`<FeatureFirstContainer /> renders with default props 1`] = `
<FeatureFirstContainer
  actions={
    Object {
      "clearDataError": [Function anonymous],
      "loadDataFailure": [Function anonymous],
      "loadDataInitiation": [Function anonymous],
      "loadDataSuccess": [Function anonymous]
    }
  }
  isLoading={false}
  store={
    Object {
      "clearActions": [Function clearActions],
      "dispatch": [Function anonymous],
      "getActions": [Function getActions],
      "getState": [Function getState],
      "replaceReducer": [Function replaceReducer],
      "subscribe": [Function subscribe]
    }
  } />
`;
